### PR TITLE
`Programming exercises`: Fix issues with hints

### PR DESCRIPTION
**Pull request by krusche**
Exercise hints have not been usable recently. This PR fixes a couple of issues so they can be used again

#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.

#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Steps for Testing

Prerequisites:
- 1 Editor
- 1 Student
- 1 Programming Exercise

1. Create a programming exercise with categories
2. Navigate into programming exercise details -> Manage hints
3. Create some hints and link them in the problem statement into tasks
4. Start the exercise as student and check if the hints are displayed properly next to the tasks

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
no changes

### Screenshots
no changes

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Hints were displayed properly.

**Review by Olimbia23 — DISMISSED** [review]
Tested on TS1. Hints are displayed properly 👍  

Screenshot 2021-11-06 at 2 15 43 PM

**Review by cremertim — DISMISSED** [review]
Tested on ts1. Everythign works as expected. I was able to see the hints

File: src/main/webapp/app/admin/system-notification-management/system-notification-management.route.ts [hunk 1]
```diff
@@ -16,7 +18,10 @@ export class SystemNotificationManagementResolve implements Resolve<any> {
      */
     resolve(route: ActivatedRouteSnapshot) {
         if (route.params['id']) {
-            return this.service.find(parseInt(route.params['id'], 10));
+            return this.service.find(parseInt(route.params['id'], 10)).pipe(
+                filter((response: HttpResponse<SystemNotification>) => response.ok),
+                map((exam: HttpResponse<SystemNotification>) => exam.body!),
```
Comment (MDK5) [inline src/main/webapp/app/admin/system-notification-management/system-notification-management.route.ts:23]: Why is this called exam? It seems to be an HttpRespone with a notification, no exams involved

**Review by MDK5 — APPROVED** [review]
Tested on ts2, was able to create and use exercise hints as editor as well as to view them as a student.

Also looked through the code and left one comment below:

**Review by rs-maker — DISMISSED** [review]
Tested and reviewed. Looks good as far as i can tell. The exercise-hint.route.ts has a 'defined but never used' issue that could be removed however it wasnt added in this PR.

File: src/main/webapp/app/admin/system-notification-management/system-notification-management.route.ts (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/admin/system-notification-management/system-notification-management.route.ts:23]: just a copy and paste error, will fix it

**Review by MDK5 — APPROVED** [review]
Retested on ts2, still works as expected, hints are usable now again

**Review by Olimbia23 — APPROVED** [review]
Re-tested on TS2. 
It works as expected. 👍

**Comment by krusche** [comment]
Thanks to all the testers and reviewers 👍

**Pull request merged**